### DISCUSSION TITLE
Cover not displayed when using map_metadata

### DISCUSSION THREAD

**Discussion by pimseb** [AzuraCast/AzuraCast#3871] [Q&A]
Hello,
I'm a new azuracast user. I've just switched from liquidsoap to azuracast.
I want to show the year of the song being played in the stream name. Until now I used this code in liquidsoap :
```def update_metadata(m) =
  title = m["title"]
  artist = m["artist"]
  year = m["year"]
  [("artist","#{artist}"),("title","#{title} [#{year}]")]
end
```
```
radio = map_metadata(update_metadata,radio)
```
I've tried to add this in the custom liquidsoap code of Azuracast. The year of the song does appear in the stream name. Unfortunately the covers disapear then (for all songs).
Is there another way to achieve what I'm trying to do ?
Thank you

**Comment by BusterNeece** [comment]
@pimseb This is because of how we detect which song is currently being played; unfortunately, Icecast and SHOUTcast both only allow very rudimentary metadata transmission, so we detect the currently playing track based on the title-artist combination. Since your metadata hook is modifying that information, we can't load any rich metadata from our system for that track, since it doesn't exactly match the track information we have in our database.

One possible workaround in your case is to add the year into the actual song title in the Media Manager, which would avoid the issue altogether.

**Comment by pimseb** [comment]
Thank you for your answer. I understand the limitation of azuracast. I will add the year in the title metadata using mp3tag.
